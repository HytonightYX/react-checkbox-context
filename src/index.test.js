import React from 'react';
import renderer from 'react-test-renderer';
import { CheckboxGroup, Checkbox } from '.';

describe('CheckboxGroup', () => {
    it('correctly sets "checked" attribute', () => {
        expect(
            renderer
                .create(
                    <CheckboxGroup name="test" initialValues={['1', '2']}>
                        <Checkbox value="1" />
                        <Checkbox value="3" />
                    </CheckboxGroup>
                )
                .toJSON()
        ).toMatchSnapshot();
    });

    it('renders correctly when deeply nested', () => {
        expect(
            renderer
                .create(
                    <CheckboxGroup name="test" initialValues={['1', '2']}>
                        <div>
                            <div>
                                <Checkbox value="1" />
                                <Checkbox value="3" />
                            </div>
                        </div>
                    </CheckboxGroup>
                )
                .toJSON()
        ).toMatchSnapshot();
    });
});
