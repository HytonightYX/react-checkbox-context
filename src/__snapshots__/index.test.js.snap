// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup correctly sets "checked" attribute 1`] = `
Array [
  <input
    checked={true}
    name="test"
    onChange={[Function]}
    type="checkbox"
    value="1"
  />,
  <input
    checked={false}
    name="test"
    onChange={[Function]}
    type="checkbox"
    value="3"
  />,
]
`;

exports[`CheckboxGroup renders correctly when deeply nested 1`] = `
<div>
  <div>
    <input
      checked={true}
      name="test"
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
    <input
      checked={false}
      name="test"
      onChange={[Function]}
      type="checkbox"
      value="3"
    />
  </div>
</div>
`;
